### PR TITLE
Added a catch for if a player is caged and kicked. Resolves #1789

### LINKED ISSUE #1789: Banning after caging
Today, I caged someone and then I banned them a minute after and they left the server but the cage was still there and I was forced to manually get rid of the glass and fix the now griefed area.

### PR DESCRIPTION
**Pull request by Wild1145**
Resolves #1789

### CODE REVIEW

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_gtfo.java [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package me.totalfreedom.totalfreedommod.command;
 
 import me.totalfreedom.totalfreedommod.banning.Ban;
+import me.totalfreedom.totalfreedommod.caging.CageData;
+import me.totalfreedom.totalfreedommod.player.FPlayer;
 import me.totalfreedom.totalfreedommod.rank.Rank;
```
Comment (OxLemonxO) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_gtfo.java]: I think you mean to put these imports in the other file...

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_gtfo.java [hunk 2]
```diff
@@ -1,6 +1,8 @@
 package me.totalfreedom.totalfreedommod.command;
 
 import me.totalfreedom.totalfreedommod.banning.Ban;
+import me.totalfreedom.totalfreedommod.caging.CageData;
+import me.totalfreedom.totalfreedommod.player.FPlayer;
```
Comment (lemonsked) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_gtfo.java]: I think you meant to put these in the other file..

**Review by JeromSar — APPROVED** [review]

File: src/main/java/me/totalfreedom/totalfreedommod/caging/Cager.java [hunk 3]
```diff
@@ -96,6 +97,18 @@ public void onPlayerQuit(PlayerQuitEvent event)
     }
 
     @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void onPlayerKick(PlayerKickEvent event)
```
Comment (lemonsked) [inline src/main/java/me/totalfreedom/totalfreedommod/caging/Cager.java:100]: Why ignoreCancelled? If the event is cancelled, why do you need to remove the cage?

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_gtfo.java (hunk 2 above)
Comment (JeromSar) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_gtfo.java]: These will need to be removed, methinks.

**Review by JeromSar — CHANGES_REQUESTED** [review]

File: src/main/java/me/totalfreedom/totalfreedommod/caging/Cager.java (hunk 3 above)
Comment (JeromSar) [reply src/main/java/me/totalfreedom/totalfreedommod/caging/Cager.java:100]: Wild's implementation is correct. If the kick event is cancelled, the player isn't kicked, so the cage data shouldn't be removed. `ignoreCancelled` causes the event handler to skip processing if the event is cancelled.

Comment (OxLemonxO) [inline src/main/java/me/totalfreedom/totalfreedommod/caging/Cager.java]: yeah but ignoreCancelled is true, causing it to ignore the event being cancelled.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_gtfo.java (hunk 2 above)
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_gtfo.java]: Yes, thats where I changed implementation mid-way through. Let me fix that.

**Comment by Wild1145** [comment]
@JeromSar Unused imports removed.

**Review by JeromSar — APPROVED** [review]

**Comment by JeromSar** [comment]
Thanks for contributing! Your PR will be merged soon.

File: src/main/java/me/totalfreedom/totalfreedommod/caging/Cager.java (hunk 3 above)
Comment (lemonsked) [reply src/main/java/me/totalfreedom/totalfreedommod/caging/Cager.java:100]: Oh, never mind then. I was thinking the other way around.

**Pull request merged**